api/v1/slo_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SloSpec defines the desired state of Slo
+type SloSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of Slo. Edit slo_types.go to remove/update
+	Name        string            `json:"name"`
+	Description string            `json:"description,omitempty"`
+	Type        string            `json:"type"`
+	Window      string            `json:"window"`
+	Slo         int               `json:"slo"`
+	Threshold   int               `json:"threshold"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"`
+}
+
+// SloStatus defines the observed state of Slo
+type SloStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	SloStatus string `json:"SloStatus"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Slo is the Schema for the sloes API
+type Slo struct {
```

Comment:
i don't think that slo is a good name. our CRD (eventually) will not only represent an slo. we need to come up with a different name. what do you think about monitor ?

Reply:
I disagree, unfortunately. SLO is a way of thinking, and you can implement it in many different ways and forms. Monitoring is a practise for understanding what your application does. Monitoring can help you in understanding and implementing your SLOs; but SLOs aren't about monitoring. They're about customers, user stories, business needs and goals, and of course - code performance and quality. 

As was discussed with Idan already a few times, we're providing an ability for engineers to easily implement SLOs manifestations as part of their code and development cycle. We've chose first to tackle SLO visualisation as Grafana Dashboards; and probably the second item would be SLO alerting as Alert Manager Alerts (which will obviously include some config in Thanos). Different ideas were also mentioned, including SLO-based-HPAs in the future - but even if we'll include those under the scope of this operator, it will still remain a SLO-centered product. So I don't think the basic item you're creating should be called "monitoring" - since you'd be creating a SLO manifestation, after understanding your client needs. The monitoring part of your code should already be there in order for you to do it. 

Even now, after saying all that, I find myself resenting the name we chose (Argos), since it comes from a myth speaking about eye sight and indeed monitoring, when this operator should be about setting objectives for your service based on data and risk assessment - but changing the whole project name is harder then keeping the basic struct.